### DISCUSSION TITLE
external ports

### DISCUSSION THREAD

**Discussion by tomflenner** [nfrastack/container-lemonldap#53] [Q&A]
hello,

do i need to use revers-proxy for nginx to run this container out of the box ? or can i expose the container port and use apache2 for exemple as reverse-proxy ?

thanks !

**Comment by tiredofit** [comment]
It _should_ be able to work if you just exposed port 80, setup the appropriate DNS records (or updated your hosts file with the manager, portal hostnames), but I wouldn't recommend it. I haven't included any TLS functionality and instead assume that the user is using something like `traefik` or `jwilder/nginx-proxy` to handle the secure bits.

**Comment by tomflenner** [comment]
where do i need to setup appropriate DNS records ?  on the docker-composer file or i can setup it directly in the docker run command ?

**Comment by tiredofit** [comment]
DNS Records would be with your DNS Provider, but you will want to set the hostnames a docker-compose file as there is 3 (4?) different web applications being served out of this one image. Check the examples folder to see how I do it, although you could also do it with `docker run -d -e ENVIRONMENTVARIABLE=value -e ANOTHERENVIRONMENTVARIABLE=value tiredofit/lemonldap`

**Comment by tomflenner** [comment]
i have a trouble with my DNS, i put MANAGER_HOSTNAME=manager.mydomain.fr but when i try to acces it i redirect me on this : "http://sso.example.com/?url=aHR0cDovL21hbmFnZXIubTFtaWFhLmZyLw%3D%3D" do you have any idea ?
